File: java/gpt/p79.java
import java.io.IOException;

public class p79 {
    public static void main(String[] args) {
        byte[] buf = new byte[1024];
        byte[] buf1 = new byte[1024];

        try {
            System.out.println("Enter first input (up to 1024 bytes):");
            int n = System.in.read(buf, 0, 1024);
            if (n > 0) {
                int copyLen = Math.min(100, n);
                System.arraycopy(buf, 0, buf1, 0, copyLen);
                System.out.println("First 100 bytes copied to buf1:");
                System.out.println(new String(buf1, 0, copyLen));
            } else {
                System.out.println("No input read for first buffer.");
            }

            // consume any leftover newline if needed
            if (System.in.available() > 0) {
                System.in.read(new byte[System.in.available()]);
            }

            System.out.println("Enter second input (up to 1024 bytes):");
            n = System.in.read(buf, 0, 1024);
            if (n > 0) {
                int copyLen = Math.min(100, n);
                System.arraycopy(buf, 0, buf1, 0, copyLen);
                System.out.println("First 100 bytes copied to buf1 from second input:");
                System.out.println(new String(buf1, 0, copyLen));
            } else {
                System.out.println("No input read for second buffer.");
            }
        } catch (IOException e) {
            System.err.println("I/O error: " + e.getMessage());
        }
    }
}

File: c/gpt/p79.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(void) {
    char *buf = malloc(1024);
    char *buf1 = malloc(1024);
    if (!buf || !buf1) {
        fprintf(stderr, "Memory allocation failed\n");
        return 1;
    }

    printf("Enter first input (up to 1024 chars):\n");
    if (fgets(buf, 1024, stdin) != NULL) {
        /* Copy the first 100 characters from buf into buf1 */
        strncpy(buf1, buf, 100);
        buf1[100] = '\0';
        printf("First 100 chars copied to buf1: %s\n", buf1);
    } else {
        printf("No input read for first buffer.\n");
    }

    printf("Enter second input (up to 1024 chars):\n");
    if (fgets(buf, 1024, stdin) != NULL) {
        /* Copy the first 100 characters from buf into buf1 (again) */
        strncpy(buf1, buf, 100);
        buf1[100] = '\0';
        printf("First 100 chars copied to buf1 from second input: %s\n", buf1);
    } else {
        printf("No input read for second buffer.\n");
    }

    free(buf);
    free(buf1);
    return 0;
}
